### DISCUSSION TITLE
Possible redundancy in handling actions?

### DISCUSSION THREAD

**Discussion by RicArch97** [labwc/labwc#2298] [General]
When i was playing a game in full screen with direct scan-out enabled, i noticed that when running LabWC with debug logging enabled, these logs were generated for every single mouse click:

```
00:00:35.665 [DEBUG] [../src/action.c:855] Handling action 25: Focus
00:00:35.665 [DEBUG] [../src/action.c:855] Handling action 29: Raise
```

In games it's pretty common for a mouse button to be pressed thousands of times, but to trigger an action in the game, not in LabWC. Especially since the window should already have focus, and already be raised, running these actions again every time might just add extra latency for no reason.

What do you think?

**Comment by Consolatis** [comment]
This is a good point, I thought we'd have some early outs for these two actions but apparently we don't. As we need to handle the mouse events anyway (to relay them to the focused surface) it isn't that much work but I think it could be reduced further.

Thoughts @jlindgren90 ?

**Reply by jlindgren90** [reply]
I highly doubt that these labwc actions are introducing latency. @RicArch97 have you done empirical comparisons with other compositors or are you just speculating?

**Reply by RicArch97** [reply]
Just speculating for now. I'll have to look into setting something up to test real world latency. Like physical mouse input to action on screen maybe.

But yeah if we really don't need to run these actions in given circumstances, might as well not do it right

**Reply by jlindgren90** [reply]
Okay, optimizations are great but let's just make sure we're looking in the right places where it will make a difference. Thanks!
